### ISSUE TITLE
Fix deprecation warning in misc.utils

### ISSUE THREAD

**Issue by tjof2** [hyperspy/hyperspy#2353]
e.g. currently reported in the tests:
```
hyperspy/tests/mva/test_bss.py::TestBSS1D::test_on_loadings
  /hyperspy/hyperspy/misc/utils.py:1143: DeprecationWarning: Using or importing
  the ABCs from 'collections' instead of from 'collections.abc' is deprecated 
  since Python 3.3,and in 3.9 it will stop working
```
Offending line looks to be:
```
    return isinstance(thing, collections.Iterable) and 
```
There may be others like this, I've not yet checked. See https://stackoverflow.com/questions/53978542/how-to-use-collections-abc-from-both-python-3-8-and-python-2-7

**Comment by tjof2** [comment]
There may be others like this, I've not yet checked

**Comment by tjof2** [comment]
Oh, this was already fixed by @ericpre in #2329 but is only available on `RELEASE_next_patch`

**Comment by ericpre** [comment]
It should be good now.
